it/uk/gov/hmrc/agentsubscriptionfrontend/connectors/MappingConnectorISpec.scala

```diff
@@ -0,0 +1,164 @@
+package uk.gov.hmrc.agentsubscriptionfrontend.connectors
+
+import java.net.URL
+
+import com.kenshoo.play.metrics.Metrics
+import uk.gov.hmrc.agentmtdidentifiers.model.Utr
+import uk.gov.hmrc.agentsubscriptionfrontend.stubs.MappingStubs
+import uk.gov.hmrc.agentsubscriptionfrontend.support.{BaseISpec, MetricTestSupport}
+import uk.gov.hmrc.http._
+
+class MappingConnectorISpec extends BaseISpec with MetricTestSupport {
+
+  private implicit val hc = HeaderCarrier()
+
+  private lazy val connector: MappingConnector =
+    new MappingConnector(
+      new URL(s"http://localhost:$wireMockPort"),
+      app.injector.instanceOf[HttpGet with HttpPost with HttpPut with HttpDelete],
+      app.injector.instanceOf[Metrics])
+
+  private def withMetricsTimerUpdate(expectedMetricName: String)(testCode: => Unit): Unit = {
```

Comment:
this would be useful in BaseISpec

Reply:
Done. Pulled it down into BaseISpec and updating all tests that check metrics to use it